docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,360 @@
     }
 
- // Execution will pause until fetchHtmlAsync finishes
- let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Parallel
+    |> Async.Ignore
+    |> Async.Start
+
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
+
+1. Transform the command-line arguments into `Async<unit>` computations with `Array.map`
+2. Create an `Async<'T[]>` that schedules and runs the `countFileBytes` computations in parallel when it runs
+3. Create an `Async<unit>` that is a composition of all work to be run, except the result is ultimately ignored
+4. Explicitly run the last computation with `Async.Start`
 
- // you actually have the result from fetchHtmlAsync now!
- printfn "%s" html
- ```
+When this program runs, `countFileBytes` runs in parallel for each command-line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
 
-2.  `Async.Start` will start an async workflow on another thread, and will **not** await its result.
+## Sequencing asynchronous computations
+
+Because `Async<'T>` is a specification of work rather than a computation that is currently running, you can perform more intricate transformations easily. For example, here is an implementation of the Sequence monad that converts a list of `Async<'T>` into an `Async<'T>` that contains an array of results. It uses the previous `countFileBytes` function:
 
 ```fsharp
-open System
-open System.Net
-  
-let uploadDataAsync url data = 
+module Async =
+    let sequence computations =
+        async {
+            let results = ResizeArray() // Where ResizeArray is the F# alias for System.Collections.List<T>
+
+            for computation in computations do
+                let! result = computation
+                results.Add(result)
+
+            return results.ToArray()
+        }
+
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
-        webClient.UploadStringAsync(uri, data)
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
-let workflow = uploadDataAsync "https://url-to-upload-to.com" "hello, world!"
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.sequence
+    |> Async.RunSynchronously
+    |> ignore
+```
+
+This will execute `countFileBytes` for each item in `argv` in the order of the elements of `argv` rather than executing them in parallel. This may be preferable if you wish to preserve the order of computations.
+
+## Common async functions
```

Comment:
Would it make sense to include `Async.Parallel` here too? It's used in one of the code examples above, but unlike the other Async functions used there it's not in this section.